sdk/src/main/java/io/dapr/client/ObjectSerializer.java

```diff
@@ -35,7 +36,8 @@ public class ObjectSerializer {
    */
   protected static final ObjectMapper OBJECT_MAPPER = new ObjectMapper()
       .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
-      .setSerializationInclusion(JsonInclude.Include.NON_NULL);
+      .configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false)
```

Comment:
This is a breaking change. Is this needed? I see that we already have a separate ObjectMapper for CloudEvent.

Reply:
I see, I just wanted to keep the OBJECT_MAPPERs in sync.

 For the CloudEvent OBJECT_MAPPER we don't need:
".configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false)"
because we are just reading data and not writing it.
On the other hand the DefaultObjectSerializer is used to read/write data and if we want the output time format to match the input time format we need that setting.
ex.
input --> "timeValue":"-999999999-01-01T00:00:00+18:00"
when serialized becomes:
output --> "timeValue":-31557014135661600.000000000"
when we don't have that option configured.

In order to support OffsetDateTime for the time fields we do need:
".findAndRegisterModules()"
in CloudEvent and ObjectSerializer, otherwise we'll get an error like the following:
"com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Java 8 date/time type java.time.OffsetDateTime not supported by default: add Module "com.fasterxml.jackson.datatype:jackson-datatype-jsr310" to enable handling".

In case we didn't want to modify the ObjectMapper at all I tried to use a field level annotation but couldn't get it to work.

ex.
"
// didn't deserialize with/without Z, and/or timezone among other setting combinations
@JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
private OffsetDateTime time;
"
or
"
// error said that this deserializer is not a constant
@JsonDeserialize(using = InstantDeserializer.OFFSET_DATE_TIME)
private OffsetDateTime time;
"

So to avoid modifying the ObjectMappers we would need to create our own CustomDeserializer ( see: https://stackoverflow.com/questions/46263773/jackson-parse-custom-offset-date-time ) and duplicate the work that the "com.fasterxml.jackson.datatype:jackson-datatype-jsr310" JavaTimeModule gave us for free.

Reply:
@artursouza I've removed the com.fasterxml.jackson.datatype:jackson-datatype-jsr310 dependency in favor of custom field level serdes for time.